orderer/common/filerepo/filerepo.go

```diff
@@ -0,0 +1,169 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package filerepo
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/hyperledger/fabric/internal/fileutil"
+	"github.com/pkg/errors"
+)
+
+const (
+	repoFilePermPrivateRW os.FileMode = 0600
+)
+
+type Repo struct {
+	mu                  sync.Mutex
+	fileRepoDir         string
+	fileSuffix          string
+	transientFileMarker string
+}
+
+// New initializes a new file repo at repoParentDir/fileSuffix.
+// All file system operations on the returned file repo are thread safe.
+func New(repoParentDir, fileSuffix string) (*Repo, error) {
+	// Check that fileSuffix doesn't contain the os path separator
+	if err := validateFileSuffix(fileSuffix); err != nil {
+		return nil, err
+	}
+
+	fileRepoDir := filepath.Join(repoParentDir, fileSuffix)
```

Comment:
Having separate subfolders helps to isolate different file repos from operating on the same folder. The suffix is still useful for filtering in case other files end up in the folder through other means.

Reply:
> Having separate subfolders helps to isolate different file repos from operating on the same folder. The suffix is still useful for filtering in case other files end up in the folder through other means.

What other means? You mean admin creates some files manually?

Reply:
Sure, I don't see it really happening but in the off chance the admin may backup a joinblock to the same directory or for some other reason accidentally move a separate file to the folder. In the end it doesn't really hurt to have the suffix in addition to the subfolder even if it introduces some minor redundancy. As for the subfolder existence itself, Yoav pointed out in a previous comment that having separate subfolders would speed up the scan and avoid filesystem sync issues if we use two file-repo's in the same process with different suffixes